Fixed #15 and #22

konsultaner opened the pull request:
@softvar I forgot to pull your changes before. You might need to update the builds again. I did not update the version, because you might want to fix things first before you release.

konsultaner commented:
@softvar since this is a breaking change you might want to update the version to 1.3.0

konsultaner commented:
> It sure why the KDF was removed as it helps strengthen the provided key. The guts of the CryptoJSWordArray.random() should have been modified to use Crypto.getRandomValues(). The current random function isn’t generating a true random number.

@jas- I completely removed the default password

- because a password should be provided by a trusted endpoint, not the browser
- because the generated password was random an needed to be saved into the localstorage `CryptoJSWordArray.random()` was only used to generate an inital password if non was provided

> You should Ben using AES, the speed up from DES shouldn’t negate use of sting ciphers

good point, ill fix this

softvar commented:
@konsultaner Could you please help in maintaining the coverage as well by adding test cases for the new code?

konsultaner commented on test/functional.spec.js:
```diff
@@ -116,7 +136,7 @@ describe('Functional tests', () => {
 
     it('should call DES encrypt if encoding is DES', () => {
-      lib = new SecureLS({encodingType: 'DES'});
+      lib = new SecureLS({encodingType: 'DES', encryptionSecret: 'some', encryptionNamespace: 'some'});
```
@softvar I did it to prevent the test from writing a lot of warnings. There is no default password anymore. Thats why a user should be warned that the encryption is insecure, but not in the test case.

konsultaner commented on src/utils.js:
```diff
@@ -17,12 +15,6 @@ let utils = {
     reason = reason ? reason : constants.WarningEnum.DEFAULT_TEXT;
     console.warn(constants.WarningTypes[reason]);
   },
-  generateSecretKey: function () {
```
Due to the issues a predefined password does not work with common security standards. You saved the password in the local storege. This was not secure. Thats why I removed it. The user should use a trusted endpoint to deliver the password. In the most cases this would be the logged in user or a trusted third party.

konsultaner commented:
> @konsultaner Could you please help in maintaining the coverage as well by adding test cases for the new code?

@softvar I mostly removed code. I added tests for the code I added. This was mainly the console warnings for `this.WarningEnum.ENCRYPTION_NAMESPACE_NOT_PROVIDED` and `this.WarningEnum.INSECURE_PASSWORD` 

If you find untested code I added let me know. I'll provide test for it then.

konsultaner commented:
@softvar I also changed some tests for the metaKey

konsultaner commented:
@softvar is there anything needed for you to accept the PR?
